### ISSUE TITLE
Add support for git subrepo

### ISSUE THREAD

**Issue by ermingol23** [magit/magit#3507]
My understanding is that Magit have support for git submodules today, alas it does not seem to have any support for git subrepo. Provided that Magit does not already have support for git subrepo, can you please add support for git subrepo as well? :)

**Comment by tarsius** [comment]
I probably won't do that. If I, or more likely someone else, does it, then it should be done in a separate package.

Unfortunately the documentation is very vague on how this is implemented and I would have to understand that before even considering to *use* it. It sounds it is more like `subtree` than `submodule`. I quite like `submodule` and think the problems that many people seem to be having with it are due to limitations of the porcelain. The basic idea and low-level implementation are solid, all that is needed is a more feature-full porcelain that caters to the needs of different kinds of users.

I like how the `subrepo` documentation mentions different kinds of users, but it does not go far enough in explaining how their needs are being address. I fear that `subrepo` is needlessly throwing out the baby with the bath water. I feel the same about `subtree` to an extend, but I do realize that there are use-cases for which it is more suitable than `submodule`. I do fear that `subrepo` is a less mature alternative to `subtree`, and since I favor `submodule` over `subtree` anyway, that puts `subrepo` in a distant third place for me.

In conclusion, I would much rather concentrate on implementing a better porcelain for `submodule` (adding some features like tracking of remotes) than adding basic support for a third `sub*` implementation.

**Comment by erichlf** [comment]
Maybe this ticket should be reopened? ...

**Comment by tarsius** [comment]
... no.

Projects comparable to Magit tend to have a few hundred open issues because that is the path of least resistance. It avoid saying upfront "No I am not going to do that" and the costs that come with that, i.e. dealing with disappointed users. I however think it is more honest and also to both the user and me to clearly communicate the decision.

If someone else wants to implement this, then that is great but until someone does, this topic does not have to sit on my public todo list. Because that is how I use the issue trackers of my projects; to say "this is what I intend to work" / "this needs to be done by me or someone else". That is not the case here.
